fe/fe-core/src/main/java/org/apache/doris/clone/TabletSchedCtx.java

```diff
@@ -499,10 +501,26 @@ public void chooseSrcReplica(Map<Long, PathSlot> backendsWorkingSlots) throws Sc
             
             long srcPathHash = slot.takeSlot(srcReplica.getPathHash());
             if (srcPathHash != -1) {
-                setSrc(srcReplica);
-                return;
+                if (!findSrcReplica) {
```

Comment:
remove the first if block

Reply:
It's necessary to record first candidate replica in the `if` block.